src/camera.nim

```diff
@@ -67,22 +67,28 @@ type Ray* = object
     tmax*: float32
     depth*: int
 
-proc newRay*(origin: Point3D, direction: Vec3f): Ray {.inline} = Ray(origin: origin, dir: direction, tmin: epsilon(float32), tmax: Inf, depth: 0)  
+proc newRay*(origin: Point3D, direction: Vec3f): Ray {.inline.} = 
+    Ray(origin: origin, dir: direction, tmin: epsilon(float32), tmax: Inf, depth: 0)  
+
 proc at*(ray: Ray; time: float32): Point3D {.inline.} = ray.origin + ray.dir * time
-proc areClose*(a, b: Ray; eps: float32 = epsilon(float32)): bool {.inline} = areClose(a.origin, b.origin, eps) and areClose(a.dir, b.dir, eps)
 
-method apply*(transf: Transformation, ray: Ray): Ray {.base, inline.} =
-    Ray(origin: apply(transf, ray.origin), dir: apply(transf, ray.dir), tmin: ray.tmin, tmax: ray.tmax, depth: ray.depth)
+proc areClose*(a, b: Ray; eps: float32 = epsilon(float32)): bool {.inline.} = 
+    areClose(a.origin, b.origin, eps) and areClose(a.dir, b.dir, eps)
 
-proc translate*(ray: Ray, vec: Vec3f): Ray {.inline.} = 
-    Ray(origin: ray.origin + vec, dir: ray.dir, tmin: ray.tmin, tmax: ray.tmax, depth: ray.depth)
+proc transform*(ray: Ray; transf: Transformation): Ray {.inline.} =
+    case transf.kind: 
+    of tkIdentity: return ray
+    of tkTranslation, tkScaling: 
+        return Ray(origin: apply(transf, ray.origin), dir: ray.dir, tmin: ray.tmin, tmax: ray.tmax, depth: ray.depth)
```

Comment:
When we use scaling i feel like we should apply it also to ray.dir. Ray direction is a vector, so can't be translated, but can still be re-normalized: if you accordingly normalize starting point and vector direction we will need the same time as before to go from the observer to a shape being part of the scenery. Isn't it better?

Reply:
I think the direction norm is not useful

Reply:
> if you accordingly normalize starting point and vector direction we will need the same time as before to go from the observer to a shape being part of the scenery. Isn't it better?
I don't understand how and when this could be used.

Reply:
If you are referring to this: 
```nim
proc at*(ray: Ray; time: float32): Point3D {.inline.} = ray.origin + ray.dir * time
```
I think the ray.dir as a direction should be used normalized, so I don't understand why a tkScaling transformation should modify also the ray direction.

Reply:
Makes sense, i thought it wasn't normalized